Feat: Migrate to Github Actions

faisalill opened the pull request:
### What's Changed:
Replace travis.yml with similar .github/workflows/tests.yml file.

### Test Plans:
Opening a PR triggers tests: 
Github Actions Test: https://github.com/faisalill/cache/pull/1
Travis Tests: https://app.travis-ci.com/github/utopia-php/cache/builds/259056517
Both have the same result.

### Mention/Reference:
https://github.com/utopia-php/framework/issues/103

faisalill committed:
Add tests.yml

faisalill committed:
Increase sleep time in tests.yml

faisalill committed:
Fix Indentatio in tests.yml

faisalill committed:
Change Docker Run Script

faisalill committed:
Change Docker Run Script

faisalill committed:
Try to enable experimental features

faisalill committed:
Try to enable experimental features

faisalill committed:
Try to enable experimental features

faisalill committed:
Try to enable experimental features

faisalill committed:
Try to enable experimental features

faisalill committed:
Change Run Tests Script

faisalill committed:
Try tests without experimental features

faisalill committed:
Run only on Pull Requests

faisalill committed:
Remove travis.yml

2002Bishwajeet commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,27 @@
+jobs:
+  tests: 
+    name: Tests
+    runs-on: ubuntu-20.04
```
```suggestion
    runs-on: ubuntu-latest
```

Let's keep it ubuntu-latest

faisalill replied:
@2002Bishwajeet 
Travis runs ubuntu-20.04. (https://github.com/utopia-php/cache/blob/master/.travis.yml#L1)
Should I change to `ubuntu-latest`???

2002Bishwajeet replied:
Yep

faisalill replied:
@2002Bishwajeet Done

2002Bishwajeet commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,27 @@
+      - name: Run Tests
+        run: |                
+            docker ps
+            docker-compose exec -T php8 vendor/bin/psalm --show-info=true
```
```suggestion
            
```
We don't need this as we already have a linter workflow for that

faisalill replied:
@2002Bishwajeet 
Isn't psalm an analyser??
We are already using phpstan analyser (https://github.com/utopia-php/cache/blob/master/composer.json#L9)

2002Bishwajeet replied:
Yep, it is. But we don't want to use this in a test workflow

faisalill replied:
@2002Bishwajeet Done

2002Bishwajeet commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,27 @@
+        run: |                
+            docker ps
+            docker-compose exec -T php8 vendor/bin/psalm --show-info=true
+            docker-compose exec -T php8 vendor/bin/phpunit --configuration phpunit.xml
```
Instead of calling this way add a script inside the `composer.json` for consistency.

For example take a look at [this](https://github.com/utopia-php/ab/blob/3ca6296f9763da30ce54fd54cfe9d1956797c9a8/composer.json#L13)

faisalill replied:
@2002Bishwajeet Done

2002Bishwajeet reviewed (changes requested):
Looks good, just some minor comments😁

faisalill committed:
Add scripts in composer.json

faisalill committed:
Make changes as per review

Use ubuntu-latest rather than ubuntu-20.04
Change Run Tests step in github actions

2002Bishwajeet reviewed (approved):
LGTM

Code QL fails but I think that's for another PR to lookout for😉

abnegate reviewed (commented):
Looks good but we need to fix the CodeQL here before we can merge

faisalill committed:
Fix CodeQL errors

faisalill committed:
Cleaner Code

faisalill committed:
Fix Lint

abnegate commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,27 @@
+name: Tests
+
+on: [pull_request]
```
Let's add on push to main as well, so merge commits are tested

faisalill committed:
Add on push to github workflow tests

abnegate commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,27 @@
+name: Tests
+
+on: [push, pull_request]
```
We only want on push when the branch is either main or master

faisalill committed:
Trigger tests on push only for main and master
